### PR TITLE
test branch to see where this will going AGAIN

### PR DESCRIPTION
**Pull request by glanttz**

### CODE REVIEW

**Comment by PixelSculptor** [comment]
> 

Did you automate creating this array of object data as I pleased yesterday? (pls say yes)

File: src/components/referrals/ImportReferralsComponent.vue [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                    v-model="nameAndSurnameDoctor">
+                    <option>Zaimportowane z dnia:</option>
+                </select>
+            </div>
+
+            <div class="inputBox">
+                <label for="researchDate"
+                    >Wybierz dzień wystawienia skierowania:</label
+                >
+                <input
```
Comment (PixelSculptor) [inline src/components/referrals/ImportReferralsComponent.vue:20]: Lack of Id for input

File: src/components/referrals/ImportReferralsComponent.vue [hunk 2]
```diff
@@ -0,0 +1,180 @@
+<template>
+    <div
+        class="referralsBox"
+        @submit.prevent="addTest">
+        <h3 class="referralsBox__header">Dodaj skierowanie:</h3>
+        <form class="referralsBox__form referralsForm">
+            <div class="inputBox">
+                <label for="doctorData"> Lekarz:</label>
```
Comment (PixelSculptor) [inline src/components/referrals/ImportReferralsComponent.vue]: delete space berfore word

File: src/components/referrals/ImportReferralsComponent.vue [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+                userId: userStore.getUserId,
+                dateImport: dateFormat.value,
+                testUrl: url.value,
+                createdAt: timestamp(),
+            });
+            if (error.value) throw new Error();
+            isLoading.value = false;
+            successFlag.value = true;
+        } catch (err) {
+            console.log(err);
```
Comment (PixelSculptor) [inline src/components/referrals/ImportReferralsComponent.vue]: why you display 3 console logs with erros

File: src/components/referrals/ImportReferralsComponent.vue [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+<script setup>
+    import ButtonComponent from '../ButtonComponent.vue';
+    import BouncingBallsComponent from '../BouncingBallsComponent.vue';
+    import ErrorInfo from '../ErrorInfo.vue';
+    import { computed, ref } from 'vue';
+    import ALLOWED_TYPES from '../../utils/allowedTypes.js';
+    import useStorage from '../../composables/getImage.js';
+    import useUserStore from '../../stores/userStore.js';
+    import { timestamp } from '../../firebase/config.js';
+    import useCollection from '../../composables/useCollections.js';
```
Comment (PixelSculptor) [inline src/components/referrals/ImportReferralsComponent.vue:67]: add space bewteen imports and group of declaring variables. its better for readable ;)

File: src/components/referrals/ReferralsComponent.vue [hunk 5]
```diff
@@ -0,0 +1,201 @@
+<template>
+    <div
+        class="referralsBox"
+        @submit.prevent="addTest">
+        <h3 class="referralsBox__header">Dodaj skierowanie:</h3>
+        <form class="referralsBox__form referralsForm">
+            <div class="inputBox">
+                <label for="doctorData"> Lekarz:</label>
```
Comment (PixelSculptor) [inline src/components/referrals/ReferralsComponent.vue:8]: You've got in ImportReferralsComponent.vue element with id = doctorData and also here. Its wrong and will be problem with styles and accessbility. You sould change it.

File: src/components/referrals/ReferralsComponent.vue [hunk 6]
```diff
@@ -0,0 +1,201 @@
...
+                <select
+                    id="doctorData"
+                    v-model="dataDoctor">
+                    <option
+                        disabled
+                        value="">
+                        Wybierz lekarza
+                    </option>
+                    <option
+                        v-for="research in getDoctorsArray()"
```
Comment (PixelSculptor) [inline src/components/referrals/ReferralsComponent.vue]: same as yesterday. Declare array without '()'. Or assign array to new variable and inject here.

File: src/components/referrals/ReferralsComponent.vue [hunk 7]
```diff
@@ -0,0 +1,201 @@
...
+    import ButtonComponent from '../ButtonComponent.vue';
+    import BouncingBallsComponent from '../BouncingBallsComponent.vue';
+    import ErrorInfo from '../ErrorInfo.vue';
+    import getDoctorsArray from '../../composables/getDoctor.js';
+    import typeOfSpecialization from '../../utils/listOfSpecialization.js';
+    import typeOfMedicalExamination from '../../utils/listOfTests.js';
+    import { computed, ref } from 'vue';
+    import { timestamp } from '../../firebase/config.js';
+    import useUserStore from '../../stores/userStore.js';
+    import useCollection from '../../composables/useCollections.js';
```
Comment (PixelSculptor) [inline src/components/referrals/ReferralsComponent.vue:100]: add spaces between imports and declaring vars

File: src/components/referrals/ReferralsComponent.vue [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+    const { isLoading, addDoc, error } = useCollection('listOfReferrals');
+    const dataDoctor = ref(null);
+    const specializations = ref(null);
+    const typeOfTestPeople = ref(null);
+    const researchDate = ref(null);
+    const successFlag = ref(false);
+    const userStore = useUserStore();
```
Comment (PixelSculptor) [inline src/components/referrals/ReferralsComponent.vue]: add space between declaring vars and defining functions or computed properties

File: src/components/referrals/ImportReferralsComponent.vue [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+    const { isLoading, addDoc, error } = useCollection('listOfImportReferrals');
+    const { url, uploadImage } = useStorage();
+    const nameAndSurnameDoctor = ref(null);
+    const researchDate = ref(null);
+    const successFlag = ref(false);
+    const testFile = ref(null);
+    const testFileError = ref(null);
+    const userStore = useUserStore();
```
Comment (PixelSculptor) [inline src/components/referrals/ImportReferralsComponent.vue:77]: add space between declaring vars and defining functions or comp properties

File: src/components/referrals/ReferralsComponent.vue [hunk 10]
```diff
@@ -0,0 +1,201 @@
...
+            await addDoc({
+                userId: userStore.getUserId,
+                date: dateFormat.value,
+                doctorData: dataDoctor.value,
+                specializationType: specializations.value,
+                testType: typeOfTestPeople.value,
+                createdAt: timestamp(),
+            });
+            if (error.value) throw new Error();
+            isLoading.value = false;
```
Comment (PixelSculptor) [inline src/components/referrals/ReferralsComponent.vue:130]: why you manually reset variable which is ref that belongs to fucntionality of useCollection hook? Look at that. If you change state isLoading ref in useCollection hook, you cannot overwrite it in this component. firstly other function control behaviour of this variable secondly: loading effect (I mean bouncing balls) will never be displayed :/

File: src/utils/listOfSpecialization.js [hunk 11]
```diff
@@ -0,0 +1,209 @@
+export const typeOfSpecialization = [
+    {
+        typeName: 'Anestezjologia',
```
Comment (PixelSculptor) [inline src/utils/listOfSpecialization.js]: Maybe change props name to: specName, specKey?

File: src/utils/listOfTests.js [hunk 12]
```diff
@@ -0,0 +1,350 @@
+export const typeOfMedicalExamination = [
+    {
```
Comment (PixelSculptor) [inline src/utils/listOfTests.js]: Maybe change props name to: medicExamName, medicExamKey?

File: src/view/ReferralsView.vue [hunk 13]
```diff
@@ -1,10 +1,136 @@
...
+    import ButtonComponent from '../components/ButtonComponent.vue';
+    import useUserStore from '../stores/userStore.js';
+    import ModalComponent from '../components/ModalComponent.vue';
+    import GreetingsComponent from '../components/GreetingsComponent.vue';
+    import UserSidebarComponent from '../components/profile/userSidebar/UserSidebarComponent.vue';
+    import ImportReferralsComponent from '../components/referrals/ImportReferralsComponent.vue';
+    import ReferralsComponent from '../components/referrals/ReferralsComponent.vue';
+
+    import { onMounted, ref } from 'vue';
+    import { onClickOutside } from '@vueuse/core';
```
Comment (PixelSculptor) [inline src/view/ReferralsView.vue:67]: as you can see here youve added yourself spaces between imports, declaring vars and functions ;)

File: src/view/ReferralsView.vue [hunk 14]
```diff
@@ -1,10 +1,136 @@
...
+        }
+        &__listOf {
+            grid-area: listof;
+            place-self: flex-start;
+        }
+        &__import {
+            grid-area: import;
+            place-self: flex-start;
+        }
+        &__userInfo {
```
Comment (PixelSculptor) [inline src/view/ReferralsView.vue:125]: why you adding style about colors etc about component that was created once?

**Review by PixelSculptor — CHANGES_REQUESTED** [review]
Add this suggestions and stay what about my questions

File: src/components/referrals/ReferralsComponent.vue (hunk 5 above)
Comment (glanttz) [reply src/components/referrals/ReferralsComponent.vue:8]: i changed id.

File: src/utils/listOfTests.js (hunk 12 above)
Comment (PixelSculptor) [reply src/utils/listOfTests.js]: And maybe stick to one name of array? I mean that youve added other file name and other name of array. Same above.

File: src/components/referrals/ReferralsComponent.vue (hunk 6 above)
Comment (glanttz) [reply src/components/referrals/ReferralsComponent.vue]: we need to have (), we talked about that yesterday. i change research to datas

Comment (PixelSculptor) [reply src/components/referrals/ReferralsComponent.vue]: Its not good notation. Better when pass data in v-for, not calling getter

File: src/components/referrals/ReferralsComponent.vue (hunk 10 above)
Comment (glanttz) [reply src/components/referrals/ReferralsComponent.vue:130]: ik, i've added missing variables

File: src/view/ReferralsView.vue (hunk 14 above)
Comment (glanttz) [reply src/view/ReferralsView.vue:125]: i will use it later, that why is created now.

Comment (PixelSculptor) [reply src/view/ReferralsView.vue:125]: No you didn't understand me. But forget about it. I have to refactor few style for usersidebar and move info about colors and border radius to exact component not above him.

File: src/components/referrals/ReferralsComponent.vue (hunk 10 above)
Comment (PixelSculptor) [reply src/components/referrals/ReferralsComponent.vue:130]: No you shouldnt add new variables. Rather delete isloading changng state if in useCollection hook isloading is changing. See yourself in definition

File: src/components/referrals/ImportReferralsComponent.vue (hunk 3 above)
Comment (PixelSculptor) [reply src/components/referrals/ImportReferralsComponent.vue]: If you import error from usecollection, you should display this error on display, and error handling in console

**Review by PixelSculptor — APPROVED** [review]

**Pull request merged**